src/features/Chat/components/ActiveWindow/MessageField.tsx

```diff
@@ -17,36 +20,53 @@ import TextInput from '@/components/TextInput';
 
 type Props = {
   chat: ChatBuddy;
-  sendMessage: (msg: NewMessage) => void;
-  isMessageSendLoading: boolean;
 };
 
-const MessageField = ({ chat, sendMessage, isMessageSendLoading }: Props) => {
+const MessageField = ({ chat }: Props) => {
   const { t } = useTranslation('chat');
   const { isMobile } = useGetLayoutMode();
-  const [text, setText] = useState('');
   const userId = useAppSelector(selectUserId);
+  const [sendMessage] = useSendMessageMutation();
+
+  const [text, setText] = useState('');
+  const [isLoadingNewMessage, setIsLoadingNewMessage] = useState(false);
 
-  const handleMessageSend = (buddyId: string, text: string) => {
-    if (!userId || isMessageSendLoading) return;
+  const handleMessageSend = async (buddyId: string, text: string) => {
+    if (!userId || isLoadingNewMessage) return;
 
+    setIsLoadingNewMessage(true);
     const message = toSendMessage(buddyId, userId, text);
-    sendMessage({ userId, message });
-    setText('');
+
+    try {
+      await sendMessage({ userId, message }).unwrap();
+    } catch (error) {
+      console.error('Failed to send message:', error);
+      // TODO: Tässä voitaisiin näyttää virheviesti käyttäjälle.
+      setIsLoadingNewMessage(false);
+    }
   };
```

Comment:
This is OK for now. I had a "better" solution in mind before.

I would like to use this lib for it https://react-hot-toast.com/
So basically we can call toasts from anywhere.
Example: https://react-hot-toast.com/docs/toast

Then we can have the toast-logic in the API-factory, and here is an example what I mean:
```typescript
// chatPageApi.ts
export const chatApi = createApi({
  baseQuery: fetchBaseQuery({ baseUrl: '/api/' }),
  reducerPath: 'chatsApi',
  endpoints: builder => ({
...
sendMessage: builder.mutation<unknown, NewMessage>({
      query: ({ userId, message }) => ({
        url: `users/${userId}/messages`,
        method: 'post',
        body: message,
      }),
      async onQueryStarted(_, {queryFulfilled }) {
        try {
           await queryFulfilled
          // `onSuccess` side-effect
          // we can do here success actions if we want
        } catch (err) {
          // `onError` side-effect
          toast.error('Error fetching post!')
        }
      },
    }),
    ```
    
    This way we dont need to have logic about the toast in the UI. 
    We can just describe the logic in the API factory. 
   This solution will scale better in the app, and it is easy to 'decorate' any query or mutation with a toast

Reply:
Yes that would be a better solution, but would require some additional restructuring. So the question is, should we prioritize it now.

Reply:
yes, we can put this on the backburner and implement this when we start to add toasts to the app

Reply:
https://trello.com/c/L2tsf4Vl/968-error-success-toasts added ticket